File: leetcode-python-go/1827. 最少操作使数组递增/main.go
package main

import (
	"fmt"
)

func minOperations(nums []int) (ans int) {
	pre := nums[0] - 1
	for _, x := range nums {
		pre = max(pre+1, x)
		ans += pre - x
	}
	return
}

func max(a, b int) int {
	if b > a {
		return b
	}
	return a
}

func main() {
	nums := []int{1, 5, 2, 4, 1}
	results := minOperations(nums)
	fmt.Println(results)
}

File: leetcode-python-go/1827. 最少操作使数组递增/main.py
from typing import List

nums = [1,5,2,4,1]

def minOperations(nums: List[int]) -> int:
    pre = nums[0] - 1
    res = 0
    for i in nums:
        pre = max(pre + 1, i)
        res += pre - i
    return res

print(minOperations(nums))